File: 1773-count-items-matching-a-rule/1773-count-items-matching-a-rule.kt
class Solution {
    fun countMatches(items: List<List<String>>, ruleKey: String, ruleValue: String): Int {
        return items.filter { when (ruleKey) {
            "color" -> it[1] == ruleValue
            "type" -> it[0] == ruleValue
            "name" -> it[2] == ruleValue
            else -> false
        } }.size
    }
}

File: 1773-count-items-matching-a-rule/1773-count-items-matching-a-rule.cpp
class Solution {
public:
int countMatches(vector<vector<string>>& items, string ruleKey, string ruleValue) {
    int count = 0;
    for (const auto &item: items) {
        if (
                (ruleKey == "type" && ruleValue == item[0]) ||
                (ruleKey == "color" && ruleValue == item[1]) ||
                (ruleKey == "name" && ruleValue == item[2])
        ) count++;
    }
    return count;
}
};